core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/deletevolumes/DeleteVolumesActions.java

```diff
@@ -0,0 +1,221 @@
+                }
+            }
+        };
+    }
+
+    @Bean(name = "DELETE_VOLUMES_FINISHED_STATE")
+    public Action<?, ?> deleteVolumesFinishedAction() {
+        return new AbstractClusterAction<>(DeleteVolumesRequest.class) {
+
+            @Override
+            protected void doExecute(ClusterViewContext context, DeleteVolumesRequest payload, Map<Object, Object> variables) {
+                StackDeleteVolumesRequest stackDeleteVolumesRequest = payload.getStackDeleteVolumesRequest();
+                StackDto stack = (StackDto) variables.get("STACK_DTO");
+                boolean detachedVolumes = false;
+                String cloudPlatform = payload.getCloudPlatform();
+                CloudPlatformVariant cloudPlatformVariant = new CloudPlatformVariant(Platform.platform(cloudPlatform), Variant.variant(cloudPlatform));
+                CloudConnector cloudConnector = cloudPlatformConnectors.get(cloudPlatformVariant);
+                CloudCredential cloudCredential = stackUtil.getCloudCredential(stack.getEnvironmentCrn());
+                Location location = location(region(stack.getRegion()), availabilityZone(stack.getAvailabilityZone()));
+                CloudContext cloudContext = CloudContext.Builder.builder()
+                        .withId(stack.getId())
+                        .withName(stack.getName())
+                        .withCrn(stack.getResourceCrn())
+                        .withPlatform(stack.getCloudPlatform())
+                        .withVariant(stack.getPlatformVariant())
+                        .withLocation(location)
+                        .withWorkspaceId(stack.getWorkspace().getId())
+                        .withAccountId(Crn.safeFromString(stack.getResourceCrn()).getAccountId())
+                        .build();
+                AuthenticatedContext ac = getAuthenticatedContext(cloudCredential, cloudContext, cloudConnector);
+                List<CloudResource> cloudResourcesToBeDeleted = payload.getResourcesToBeDeleted();
+                try {
+                    deleteVolumesService.detachResources(cloudResourcesToBeDeleted, cloudPlatform, ac);
+                    deleteVolumesService.updateCbdbResources(stack, payload);
+                    detachedVolumes = true;
+                    DeleteVolumesFinalizedEvent event = new DeleteVolumesFinalizedEvent(payload.getStackDeleteVolumesRequest().getStackId());
+                    sendEvent(context, event);
+                } catch (Exception ex) {
+                    LOGGER.error("Detaching EBS disks failed for stack: {}, and group: {}, Exception:: {}", stackDeleteVolumesRequest.getStackId(),
+                            stackDeleteVolumesRequest.getGroup(), ex.getCause());
+                    DeleteVolumesResult deleteVolumesResult = new DeleteVolumesResult(ex.getMessage(), ex, stack.getId());
+                    sendEvent(context, CloudPlatformResult.failureSelector(DeleteVolumesResult.class), deleteVolumesResult);
+                }
```

Comment:
Even if detaching volumes is successful but sending event is failed , FAIL_HANDLED_EVENT will be sent. Since detachedVolumes is true, it will still go ahead with deletion .